### PR TITLE
[webui] Add link 'more info' in package flash error

### PR DESCRIPTION
**Pull request by DavidKang**
before
screenshot from 2017-03-03 18-16-57

after
screenshot from 2017-03-07 11-31-22

showing more info
screenshot from 2017-03-07 11-36-43

### CODE REVIEW

File: src/api/app/assets/javascripts/webui/application/package.js [hunk 1]
```diff
@@ -145,3 +145,17 @@ function live_build_log_ready() {
     $('.start_refresh').click(start_refresh);
     $('.stop_refresh').click(stop_refresh);
 }
+
+$( document ).ready(function() {
+  $('.btn-more').click(function() {
+    var link = $(this);
+    $('.more_info').toggle(0, function() {
+      if ($(this).is(':visible')) {
```
Comment (Ana06) [inline src/api/app/assets/javascripts/webui/application/package.js]: `if (link.is(':visible'))`

File: src/api/app/assets/javascripts/webui/application/package.js [hunk 2]
```diff
@@ -145,3 +145,17 @@ function live_build_log_ready() {
     $('.start_refresh').click(start_refresh);
     $('.stop_refresh').click(stop_refresh);
 }
+
+$( document ).ready(function() {
+  $('.btn-more').click(function() {
+    var link = $(this);
+    $('.more_info').toggle(0, function() {
+      if (link.is(':visible')) {
```
Comment (Ana06) [inline src/api/app/assets/javascripts/webui/application/package.js]: what about writing it like this:

`$(this).is(':visible') ? link.text('less info') : link.text('more info');`

File: src/api/spec/support/files/service [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<services>
```
Comment (ChrisBr) [inline src/api/spec/support/files/service]: I would rename the file to broken service or something! It is on a first glance not clear that this is broken! Also is it really necessary to have a full file for this? Wouldn't it be enough to just have ``foobar`` as content for a broken service?

**Comment by adrianschroeter** [comment]
hm, the full log can become long (imaging a build log file), is that still working with that view?

The \n should become real new-lines also. Is that an api or backend problem?

File: src/api/spec/support/files/service (hunk 3 above)
Comment (DavidKang) [inline src/api/spec/support/files/service]: Right! I will rename the file. And yes it's not necessary to have a full file, in fact, we don't need the file, so I will remove it and add a little content.

**Comment by DavidKang** [comment]
With full log we have to do scroll to see all the content. I will convert \n in new-lines

File: src/api/spec/controllers/webui/package_controller_spec.rb [hunk 4]
```diff
@@ -424,6 +425,16 @@
 
       it { expect(assigns(:linking_packages)).to match_array(package.linking_packages) }
     end
+
+    context 'with a package that have a broken service' do
```
Comment (bgeuken) [inline src/api/spec/controllers/webui/package_controller_spec.rb]: \*that has

File: src/api/spec/features/webui/packages_spec.rb [hunk 5]
```diff
@@ -175,4 +175,20 @@
...
+      login user
+      visit package_show_path(package: broken_package, project: user.home_project)
+    end
+
+    scenario 'show more info' do
+      click_link('more info')
+      expect(find('#flash-messages').text).not_to be_empty
+      expect(page).to have_css(".more_info")
+      expect(find('.more_info').text).not_to be_empty
+      expect(find(".btn-more").text).to eq('less info')
```
Comment (bgeuken) [inline src/api/spec/features/webui/packages_spec.rb]: I think it's worth to test that we actually show the content of the error / expanded error

File: src/api/app/models/package.rb [hunk 6]
```diff
@@ -897,6 +897,11 @@ def buildresults(prj = project)
     local_build_results
   end
 
+  def service_error(revision = nil)
+    revision ||= serviceinfo.to_hash['xsrcmd5']
+    PackageServiceErrorFile.new(project_name: project.name, package_name: name).to_s({ rev: revision })
```
Comment (evanrolfe) [inline src/api/app/models/package.rb]: Brackets aren't needed here, can just do: 
```ruby
to_s(rev: revision)
```

File: src/api/spec/features/webui/packages_spec.rb (hunk 5 above)
Comment (evanrolfe) [inline src/api/spec/features/webui/packages_spec.rb]: Didn't we say in the conversation yesterday that feature specs should only be used for work flows?

Comment (bgeuken) [inline src/api/spec/features/webui/packages_spec.rb]: Still we should test the actual outcome. Where you want to do that I don't mind.

File: src/api/app/models/package.rb [hunk 7]
```diff
@@ -897,6 +897,11 @@ def buildresults(prj = project)
     local_build_results
   end
 
+  def service_error(revision = nil)
```
Comment (hennevogel) [inline src/api/app/models/package.rb]: What happens if there are no serivces?

Comment (DavidKang) [inline src/api/app/models/package.rb]: Now, return nil

**Comment by DavidKang** [comment]
@evanrolfe, when you can review it, please

File: src/api/spec/features/webui/packages_spec.rb [hunk 8]
```diff
@@ -175,4 +175,21 @@
       expect(page.source).to have_text('[1] this is my dummy logfile')
     end
   end
+
+  describe "with 'more info' link in flash error" do
```
Comment (evanrolfe) [inline src/api/spec/features/webui/packages_spec.rb]: Why is there still this feature spec? They are only for testing workflows..

**Comment by evanrolfe** [comment]
LGTM (assuming travis passes)

**Pull request merged**